docs/standard/serialization/snippets/system-text-json-how-to/csharp/RoundtripDataTable.cs

```diff
@@ -72,66 +72,33 @@ public override void Write(Utf8JsonWriter jsonWriter, DataTable value, JsonSeria
-                    {
-                        jsonWriter.WriteNumber(key, (uint)dr[col]);
-                    }
-                    else if (col.DataType == typeof(ulong))
-                    {
-                        jsonWriter.WriteNumber(key, (ulong)dr[col]);
-                    }
-                    else if (col.DataType == typeof(DateTime))
-                    {
-                        jsonWriter.WriteString(key, (DateTime)dr[col]);
-                    }
-                    else if (col.DataType == typeof(Guid))
-                    {
-                        jsonWriter.WriteString(key, (Guid)dr[col]);
-                    }
-                    else
-                    {
-                        jsonWriter.WriteString(key, dr[col].ToString());
-                    }
+                    Action<string> action = GetWriteAction(dr, col, jsonWriter);
+                    action.Invoke(key);
+
+                    static Action<string> GetWriteAction(
+                        DataRow row, DataColumn column, Utf8JsonWriter writer) => column.DataType switch
+                        {
+                            // bool
+                            _ when column.DataType == typeof(bool) => key => writer.WriteBoolean(key, (bool)row[column]),
+
+                            // numbers
+                            _ when column.DataType == typeof(byte) => key => writer.WriteNumber(key, (byte)row[column]),
+                            _ when column.DataType == typeof(sbyte) => key => writer.WriteNumber(key, (sbyte)row[column]),
+                            _ when column.DataType == typeof(decimal) => key => writer.WriteNumber(key, (decimal)row[column]),
+                            _ when column.DataType == typeof(double) => key => writer.WriteNumber(key, (double)row[column]),
+                            _ when column.DataType == typeof(float) => key => writer.WriteNumber(key, (float)row[column]),
+                            _ when column.DataType == typeof(short) => key => writer.WriteNumber(key, (short)row[column]),
+                            _ when column.DataType == typeof(int) => key => writer.WriteNumber(key, (int)row[column]),
+                            _ when column.DataType == typeof(ushort) => key => writer.WriteNumber(key, (ushort)row[column]),
+                            _ when column.DataType == typeof(uint) => key => writer.WriteNumber(key, (uint)row[column]),
+                            _ when column.DataType == typeof(ulong) => key => writer.WriteNumber(key, (ulong)row[column]),
+
+                            // strings
+                            _ when column.DataType == typeof(DateTime) => key => writer.WriteString(key, (DateTime)row[column]),
+                            _ when column.DataType == typeof(Guid) => key => writer.WriteString(key, (Guid)row[column]),
+
+                            _ => key => writer.WriteString(key, row[column].ToString())
+                        };
```

Comment:
Would switching over `row[column]` with each switch arm being simplified to, for example, `bool b`, then use the "b" instead of a cast work? If it works, I think it will look nicer?

Reply:
Note: I wrote this directly in GitHub, so I may have an error.

```suggestion
                        DataRow row, DataColumn column, Utf8JsonWriter writer) => row[column] switch
                        {
                            // bool
                            bool value => key => writer.WriteBoolean(key, value),

                            // numbers
                            byte value => key => writer.WriteNumber(key, value),
                            sbyte value => key => writer.WriteNumber(key, value),
                            decimal value => key => writer.WriteNumber(key, value),
                            double value => key => writer.WriteNumber(key, value),
                            float value => key => writer.WriteNumber(key, value),
                            short value => key => writer.WriteNumber(key, value),
                            int value => key => writer.WriteNumber(key, value),
                            ushort value => key => writer.WriteNumber(key, value),
                            uint value => key => writer.WriteNumber(key, value),
                            ulong value => key => writer.WriteNumber(key, value),

                            // strings
                            DateTime value => key => writer.WriteString(key, value),
                            Guid value => key => writer.WriteString(key, value),

                            { } value => key => writer.WriteString(key, value)
                        };
```

Reply:
@Youssef1313 Your code works correctly, but the last line needs to change to 

```
_ => key => writer.WriteString(key, row[column].ToString())
```